radix-engine-interface/src/math/decimal.rs

```diff
@@ -307,11 +307,12 @@ where
     type Output = Decimal;
 
     fn mul(self, other: T) -> Self::Output {
-        let a = self.0;
+        let a = I320::from(self.0);
         let b_dec: Decimal = other.try_into().expect("Overflow");
-        let b: I256 = b_dec.0;
-        let c: I256 = a * b / Self::ONE.0;
-        Decimal(c)
+        let b: I320 = I320::from(b_dec.0);
+        let c: I320 = a * b / I320::from(Self::ONE.0);
```

Comment:
We should promote to `I512` instead of `I320`, otherwise `a * b` will overflow.

Reply:
After more thoughts, `I320` should also be fine. If it were to overflow `a * b`, we won't be able to convert it back to `Decimal` anywa.